pkg/services/secrets/kvstore/fallback.go

```diff
@@ -0,0 +1,67 @@
+package kvstore
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type KVStoreWithFallback struct {
+	log      log.Logger
+	store    SecretsKVStore
+	fallback SecretsKVStore
+}
+
+func NewKVStoreWithFallback(store SecretsKVStore, fallback SecretsKVStore) *KVStoreWithFallback {
+	return &KVStoreWithFallback{
+		log:      log.New("secrets.kvstore"),
+		store:    store,
+		fallback: fallback,
+	}
+}
+
+func (kv *KVStoreWithFallback) Get(ctx context.Context, orgId int64, namespace string, typ string) (string, bool, error) {
+	value, ok, err := kv.store.Get(ctx, orgId, namespace, typ)
+	if err != nil || !ok {
```

Comment:
we might want to guard this a bit more with a check for whether a migration is actually occurring. If we already migrated and then there is an error or exists = false, then we should probably just return the error. Maybe the migrator can set an in-memory flag after it has completed and we can check if it has been set here?

Reply:
In this case the fallback secret store could be used only for migration, so it shouldn't be returned on kvstore.go

Reply:
Added a flag to the struct that migration can use to disable fallback once it's done.